src/sentry/issues/endpoints/group_similar_issues_embeddings.py

```diff
@@ -40,22 +41,30 @@ class GroupSimilarIssuesEmbeddingsEndpoint(GroupEndpoint):
         "GET": ApiPublishStatus.PRIVATE,
     }
 
+    def get_group_hashes_for_group_id(self, project_id: int, group_id: int) -> set[str]:
+        hashes = GroupHash.objects.filter(project_id=project_id, group_id=group_id)
```

Comment:
I don't think you need the project ID. I could be wrong though.

Reply:
no harm in keeping it though

Reply:
thanks all! I'll remove it just to make it cleaner

Reply:
As someone who's had to spend a bunch of time onboarding to the grouping codebase, I actually disagree. Anything which muddies the waters is something I'd argue in favor of avoiding.

Reply:
Interesting.

In this case since what this is doing is actively _not_ showing information, that seems not to apply, though, no?

Regardless, I'm not totally clear on what you're suggesting - that we should always include project id whenever we query the database on group id? That feels like it covers a _lot_ of our code. That said, I also don't totally get how doing that mitigates the kind of attack the linked article talks about. Is the idea just that if I'm a bad guy randomly guessing group ids, I'm unlikely to randomly guess the corresponding project id?

Reply:
Also, on the topic of auth, don't all of our endpoints require an API key to access?

Reply:
>Regardless, I'm not totally clear on what you're suggesting - that we should always include project id whenever we query the database on group id? That feels like it covers a lot of our code. That said, I also don't totally get how doing that mitigates the kind of attack the linked article talks about. Is the idea just that if I'm a bad guy randomly guessing group ids, I'm unlikely to randomly guess the corresponding project id?

yeah, it in theory should. but i will digress and bring this up in a TSC setting -- no need to think about for now. 

>Also, on the topic of auth, don't all of our endpoints require an API key, with particular scopes, to access, that would prevent random bad guys from getting at stuff?

no. imagine you have a called ProjectSomeThingy endpoint which is subclassed from `ProjectEndpoint`. the endpoint's path is `/project/{org}/{project}/somethingy/{id}`. The endpoint will return a serialized `SomeThingy` based on the id you pass in. The `ProjectEndpoint` validates that you belong to your project, but it *does not* validate that you own the SomeThingy object.

So unless you include `project_id` in your query to the `SomeThingy` table, you can in theory access any project's `SomeThingy`. I would encourage folks to read https://cheatsheetseries.owasp.org/cheatsheets/Insecure_Direct_Object_Reference_Prevention_Cheat_Sheet.html as this is described in detail there. (in this endpoints case we use A GroupEndpoint which ensures that this Group is accociated with the project, so we are safe)